napari/tests/test_viewer.py

```diff
@@ -54,254 +55,41 @@ def test_no_qt_loop():
-    assert np.sum(view.dims._displayed_sliders) == 0
-
-    # Switch to 3D rendering mode and back to 2D rendering mode
-    viewer.dims.ndisplay = 3
-    assert viewer.dims.ndisplay == 3
-    viewer.dims.ndisplay = 2
-    assert viewer.dims.ndisplay == 2
-
-    # Run all class keybindings
-    for func in layer.class_keymap.values():
-        func(layer)
-
-    # Close the viewer
-    viewer.window.close()
-
-
-def test_add_vectors(qtbot):
-    """Test adding vectors."""
-    viewer = Viewer()
-    view = viewer.window.qt_viewer
-    qtbot.addWidget(view)
-
-    np.random.seed(0)
-    data = 20 * np.random.random((10, 2, 2))
-    layer = viewer.add_vectors(data)
-    assert np.all(viewer.layers[0].data == data)
-
-    assert len(viewer.layers) == 1
-    assert view.layers.vbox_layout.count() == 2 * len(viewer.layers) + 2
-
-    assert viewer.dims.ndim == 2
-    assert view.dims.nsliders == viewer.dims.ndim
-    assert np.sum(view.dims._displayed_sliders) == 0
-
-    # Switch to 3D rendering mode and back to 2D rendering mode
-    viewer.dims.ndisplay = 3
-    assert viewer.dims.ndisplay == 3
-    viewer.dims.ndisplay = 2
-    assert viewer.dims.ndisplay == 2
-
-    # Run all class keybindings
-    for func in layer.class_keymap.values():
-        func(layer)
-
-    # Close the viewer
-    viewer.window.close()
-
-
-def test_add_shapes(qtbot):
-    """Test adding shapes."""
-    viewer = Viewer()
-    view = viewer.window.qt_viewer
-    qtbot.addWidget(view)
-
-    np.random.seed(0)
-    data = 20 * np.random.random((10, 4, 2))
-    layer = viewer.add_shapes(data)
-    assert np.all(viewer.layers[0].data == data)
-
-    assert len(viewer.layers) == 1
-    assert view.layers.vbox_layout.count() == 2 * len(viewer.layers) + 2
-
-    assert viewer.dims.ndim == 2
-    assert view.dims.nsliders == viewer.dims.ndim
-    assert np.sum(view.dims._displayed_sliders) == 0
-
-    # Switch to 3D rendering mode and back to 2D rendering mode
-    viewer.dims.ndisplay = 3
-    assert viewer.dims.ndisplay == 3
-    viewer.dims.ndisplay = 2
-    assert viewer.dims.ndisplay == 2
-
-    # Run all class keybindings
-    for func in layer.class_keymap.values():
-        func(layer)
-
-    # Close the viewer
-    viewer.window.close()
-
-
-def test_add_surface(qtbot):
-    """Test adding 3D surface."""
-    viewer = Viewer()
-    view = viewer.window.qt_viewer
-    qtbot.addWidget(view)
-
-    np.random.seed(0)
-    vertices = np.random.random((10, 3))
-    faces = np.random.randint(10, size=(6, 3))
-    values = np.random.random(10)
-    data = (vertices, faces, values)
-    layer = viewer.add_surface(data)
-    assert np.all(
-        [np.all(vd == d) for vd, d in zip(viewer.layers[0].data, data)]
-    )
-
-    assert len(viewer.layers) == 1
+    if layer_type is LayerTypes.SURFACE:
+        assert np.all(
+            [np.all(vd == d) for vd, d in zip(viewer.layers[0].data, data)]
+        )
+    else:
+        assert len(viewer.layers) == 1
```

Comment:
This line should be True for all layer types, so doesn't need to be in the `else` clause